### PR TITLE
[BartTokenizerFast] add prepare_seq2seq_batch

### PR DESCRIPTION
**Pull request by patil-suraj**
This PR adds  `prepare_seq2seq_batch` batch method for `BartTokenizerFast` as per the proposal in #6080

@sshleifer

### CODE REVIEW

File: tests/test_modeling_bart.py [hunk 1]
```diff
@@ -634,6 +639,76 @@ def test_special_tokens(self):
...
+        self.assertEqual(32, batch["decoder_attention_mask"].shape[1])
+
+    def test_outputs_not_longer_than_maxlen_fast(self):
+        tokenizer = self.default_tokenizer_fast
+
+        batch = tokenizer.prepare_seq2seq_batch(["I am a small frog" * 1024, "I am a small frog"], return_tensors="pt")
+        self.assertIsInstance(batch, BatchEncoding)
+        self.assertEqual(batch.input_ids.shape, (2, 1024))
+
+    def test_special_tokens_fast(self):
```
Comment (patil-suraj) [inline tests/test_modeling_bart.py:698]: @sshleifer  just copy pasted the tests, any better way for this ?

Comment (sshleifer) [reply tests/test_modeling_bart.py:698]: you could use a for loop or inheritance, but this is fine.

**Review by sshleifer — APPROVED** [review]

File: tests/test_modeling_bart.py [hunk 2]
```diff
@@ -570,144 +570,82 @@ def test_cnn_summarization_same_as_fairseq(self):
         # TODO(SS): add test case that hits max_length
 
     def test_prepare_seq2seq_batch(self):
-        tokenizer = self.default_tokenizer
+        tokenizers = [self.default_tokenizer, self.default_tokenizer_fast]
```
Comment (patil-suraj) [inline tests/test_modeling_bart.py:573]: @sshleifer , used for loop and tested both tokenizers in same method

Comment (sshleifer) [reply tests/test_modeling_bart.py:573]: nice.

**Review by sgugger — APPROVED** [review]
LGTM, thanks for adding this!

**Pull request merged**